app/src/main/java/org/fossasia/susi/ai/chat/ChatPresenter.kt

```diff
@@ -457,59 +469,85 @@ class ChatPresenter(context: Context) :
             val actionSize = susiResponse.answers[0].actions.size
             val date = susiResponse.answerDate
 
-            for (i in 0 until actionSize) {
-                val delay = susiResponse.answers[0].actions[i].delay
-                val handler = Handler()
-                handler.postDelayed({
-                    val parseSusiHelper = ParseSusiResponseHelper()
-                    parseSusiHelper.parseSusiResponse(susiResponse, i, utilModel.getString(R.string.error_occurred_try_again))
-
-                    var setMessage = parseSusiHelper.answer
-                    if (parseSusiHelper.actionType == Constant.TABLE) {
-                        tableItem = parseSusiHelper.tableData
-                    } else if (parseSusiHelper.actionType == Constant.VIDEOPLAY || parseSusiHelper.actionType == Constant.AUDIOPLAY) {
-                        // Play youtube video
-                        identifier = parseSusiHelper.identifier
-                        chatView?.playVideo(identifier)
-                    } else if (parseSusiHelper.actionType == Constant.ANSWER && (PrefManager.checkSpeechOutputPref() && check || PrefManager.checkSpeechAlwaysPref())) {
-                        setMessage = parseSusiHelper.answer
-
-                        var speechReply = setMessage
-                        if (parseSusiHelper.isHavingLink) {
-                            speechReply = setMessage.substring(0, setMessage.indexOf("http"))
+            for (j in 0 until actionSize) {
+                val parseSusiHelper = ParseSusiResponseHelper()
```

Comment:
Why allocating in a loop?

Reply:
The actions part of the susi response contains a set of replies. Each of these replies needs to be parsed individually singr PareseSusiResponseHelper class. As it is necessary to identify the action type, whether it is a video or something else.